test/server/cards/01_SOR/IAmYourFather.spec.js

```diff
@@ -0,0 +1,50 @@
+describe('I Am Your Father', function() {
+    integration(function() {
+        describe('I Am Your Father\'s ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['i-am-your-father'],
+                        deck: ['foundling', 'pyke-sentinel', 'atst', 'cartel-spacer', 'battlefield-marine'],
+                        groundArena: ['wampa'],
+                    },
+                    player2: {
+                        groundArena: ['viper-probe-droid'],
+                        leader: { card: 'darth-vader#dark-lord-of-the-sith', deployed: true }
+                    }
+                });
+            });
+
+            it('can only select opponent\'s units, including leaders', function () {
+                this.player1.clickCard(this.iAmYourFather);
+                expect(this.player1).toBeAbleToSelectExactly([this.viperProbeDroid, this.darthVader]);
+            });
+
+            it('gives the opponent a choice after selecting the target unit', function () {
```

Comment:
Since there is a lot of setup for each test case (each case has to do the entire game setup), we try to consolidate tests as much as possible. So in this case instead of having a test which confirms each stage of the process, I would prefer consolidating down to 2 cases (one for each button) in which one or both tests have expectations regarding the right targets being available

Reply:
Got it. I figured each test should be testing exactly one thing as much as possible(the thing you describe in the first argument to it()), but I can condense the tests down.

Reply:
Yeah, I think ideally that would be the case but our test infra kind of sucks and we have a finite amount of github actions compute time available so we need to condense a bit